Create io.elementary.BaseApp.yml

bilelmoussaoui opened the pull request:
A lot of elementary apps are now upgraded to use Juno's stack. Instead of upgrading the Loki baseApp, creating a new will allow keeping the non-updated apps working too.
Currently, the elementary Gtk & icons don't provide any correct tags at all :( So, I went with the latest "stable" commit and will try to contact upstream about that.

jurf commented:
Hm, why not just do a `io.elementary.BaseApp`, and have multiple branches? Like QtWebKit does currently.

bilelmoussaoui commented:
I wasn't aware of that. We have already pushed the Loki BaseApp and it's not possible to update to change the AppID it's quite impossible to do that right now...

jurf commented:
Not really, we don’t need AppID changing capabilities any more as if we’ve been changing the runtime, just the existing apps relying on Loki need to have their manifest updated and then the .Loki base-app can be EoL-ed.

bilelmoussaoui commented:
Oh I see. I can do that for Juno and then push a Loki branch & update the loki apps either to the new baseApp or to point to the correct baseApp

nedrichards commented:
yep, that would work. for baseapp's in particular we control both pieces.

nedrichards commented:
bot, build io.elementary.Juno.BaseApp

bilelmoussaoui commented:
Why does this happen while building libgee 
```
 /usr/bin/mkdir -p '/usr/share/gir-1.0'
 /usr/bin/install -c -m 644 Gee-0.8.gir '/usr/share/gir-1.0'
/usr/bin/install: cannot create regular file '/usr/share/gir-1.0/Gee-0.8.gir': Read-only file system
make[2]: *** [Makefile:1267: install-dist_girDATA] Error 1
make[2]: Leaving directory '/run/build/libgee/gee'
make[1]: *** [Makefile:1446: install-am] Error 2
make[1]: Leaving directory '/run/build/libgee/gee'
make: *** [Makefile:503: install-recursive] Error 1
```
The same version of libgee works out of the box on the Loki BaseApp. Any ideas?

jurf commented:
Looks like a problem with the prefix. 18.08 might be at fault. Try passing the relevant flag manually.

nedrichards commented:
I believe it's because the env var stuff that we've been hacking for 1.6 onwards doesn't work in the new package config that's available in 18.08. All stuff that does that will need to change.

jurf commented:
@nedrichards you sure about that? `cannot create regular file '/usr/share/gir-1.0/Gee-0.8.gir'` really looks like a wrong/no prefix was passed.

bilelmoussaoui commented:
@jurf I agree with @nedrichards. Passing `--prefix` doesn't resolve the issue.  The path for the gir files are set using 
```
    "build-options": {
        "env": {
            "PKG_CONFIG_GOBJECT_INTROSPECTION_1_0_GIRDIR": "/app/share/gir-1.0",
            "PKG_CONFIG_GOBJECT_INTROSPECTION_1_0_TYPELIBDIR": "/app/lib/girepository-1.0"
        }
    },
```
Which doesn't work on the new FreeDesktop 18.08 runtime anymore :/

TingPing commented:
Another package was passing `girdir=foo` or such to `make-install-args`.

bilelmoussaoui commented:
@TingPing Using this with libgee 
```
      - type:         shell
        commands:
          - sed -i 's/`$PKG_CONFIG --variable=girdir gobject-introspection-1.0`/\/app\/share\/gir-1.0/g' configure
          - sed -i 's/"$($PKG_CONFIG --variable=typelibdir gobject-introspection-1.0)"/\/app\/lib\/girepository-1.0/g' configure

```
does the work. But again, it fails with the same issue on granite :/ Which meant it has to be applied to every module out there that ships a gir or a typelib file! Will try to see what FreeDestkop guys thinks about it

bilelmoussaoui commented:
Ok, fixed the build of libgee & granite and updated to using io.elementary.BaseApp. This should be pushed to a branch called `Juno` and will add a `Loki` branch once this one is merged. Will updated the rest of the apps later before making the old BaseApp EoL

nedrichards commented:
bot, build io.elementary.BaseApp

bilelmoussaoui commented:
Builds successfully https://flathub.org/builds/#/builders/40/builds/453. Will try to see if libgee maintainers (if there any) are interested in switching to meson

barthalion commented:
We totally can switch to one repo with multiple branches if you don't plan to change runtime version in between Elementary releases.

bilelmoussaoui commented:
So all the versions of the baseAop has to use the same version of the runtime? Well it can work for now but not sure about the future...

barthalion commented:
No, each version of the base app should keep using the one it has been added with. Otherwise we break ABI stability for maintainers.

bilelmoussaoui commented:
Yeah that what I'm willing to do:)the versions are based on what elementary ships by default so nothing will change until the next major release

bilelmoussaoui commented:
Is there anything else that should be done to get this one merged? They are a bunch of elementary devs that are willing to distribute their app on Flathub, this will allow me to give a hand for the Juno only apps.

Alexander-Wilms commented:
libgranite.so gets installed to /app/lib64 instead of /app/lib, so e.g. Agenda's cmake configuration doesn't find it by default.

bilelmoussaoui commented:
@Alexander-Wilms Should be fixed in my latest commit.

Alexander-Wilms commented:
Thanks

nedrichards commented:
bot, build io.elementary.BaseApp

bilelmoussaoui commented:
Builds successfully https://flathub.org/builds/#/builders/2/builds/5221

Alexander-Wilms commented:
Have you tested elementary applications built with this BaseApp? The theme is broken in Agenda, the window is completely transparent.

bilelmoussaoui commented:
@Alexander-Wilms Which version of Agenda are you using? It's a commit from the master branch until there's a stable release

bilelmoussaoui commented:
@Alexander-Wilms Have you patched Agenda to use the elementary Gtk theme?

bilelmoussaoui commented:
Anyway, this one is ready. For the granite installing the lib in /lib64 instead of /lib it's an issue with the freedesktop runtime not carrying a patch to force using /lib, it was reported and there's a MR to fix it. Fixed that here :)

TingPing commented on io.elementary.BaseApp.yml:
```diff
@@ -0,0 +1,72 @@
+id:                   io.elementary.BaseApp
```
Don't do this weird indenting, nothing else does it.

jurf replied:
I’m actually planning to switch my apps to this indenting, it looks so much cleaner IMO.

bilelmoussaoui replied:
Just used the default atom-beautify plugin, I can revert that but it looks really clean

jurf replied:
Ah yeah, didn’t think about the diff.

TingPing replied:
I created an entry just for this conversation: https://github.com/flathub/flathub/wiki/YAML-Style-Guide

(I assume other maintainers will agree)

bilelmoussaoui replied:
@TingPing Fixed.

bilelmoussaoui commented:
Anything else I should do before getting this ready? This will allow me to fix a bunch of apps from #667

TingPing commented on io.elementary.BaseApp.yml:
```diff
@@ -0,0 +1,72 @@
+      - /bin
+      - /share/applications
+    sources:
+      - type:         archive
```
Missed de-indenting these.

bilelmoussaoui replied:
Fixed that.

barthalion commented:
Repository has been created: https://github.com/flathub/io.elementary.BaseApp

Thanks!
